test/TestUtilities/Test.Utility/TestServer/TcpListenerServer.cs

```diff
@@ -41,6 +49,11 @@ public async Task<T> ExecuteAsync<T>(Func<string, Task<T>> action)
                     var serverTask = startServer(tcpListener, serverCts.Token);
                     var address = $"http://localhost:{port}/";
 
+                    if (Mode == TestServerMode.InvalidTLSCertificate)
+                    {
+                        address = $"https://localhost:{port}/";
+                    }
```

Comment:
This might be easier to read if expressed as an if/else rather than an assignment then an if with an override of that assignment.

Either as below or using a ternary operator for the assignment:

```suggestion
                    if (Mode == TestServerMode.InvalidTLSCertificate)
                    {
                        address = $"https://localhost:{port}/";
                    }
                    else
                    {
                        var address = $"http://localhost:{port}/";
                    }
```